src/menu/create_wallet_menu.c

```diff
@@ -179,13 +193,36 @@ static void create_wallet_menu_handler(engine_ctx_t *ctx,
     // UI_EVENT_LIST_REJECTION handled below already
   }
 
+  if (NULL != flash_wallet && ((VALID_WALLET != flash_wallet->state) ||
+                               (0x0F != flash_wallet->cards_states))) {
+    handle_wallet_creation_failure(flash_wallet);
+  }
+
   /* Return to the previous menu irrespective if UI_EVENT_REJECTION was
    * detected, or a create wallet flow was executed */
   engine_delete_current_flow_step(ctx);
 
   return;
 }
 
+void handle_wallet_creation_failure(Flash_Wallet *flash_wallet) {
+  if (NULL == flash_wallet) {
+    return;
+  }
```

Comment:
can remove this duplicate check. it is already checked before calling this function

Reply:
removed